searchcore/src/vespa/searchcore/proton/server/maintenance_jobs_injector.cpp

```diff
@@ -34,15 +36,27 @@ injectLidSpaceCompactionJobs(MaintenanceController &controller,
                              const std::shared_ptr<IBucketStateCalculator> &calc,
                              document::BucketSpace bucketSpace)
 {
-    (void) bucketSpace;
     for (auto &lidHandler : lscHandlers) {
-        IMaintenanceJob::UP job = std::make_unique<LidSpaceCompactionJob>(
-                config.getLidSpaceCompactionConfig(),
-                *lidHandler, opStorer, fbHandler,
-                diskMemUsageNotifier,
-                config.getBlockableJobConfig(),
-                clusterStateChangedNotifier,
-                (calc ? calc->nodeRetired() : false));
+
+        IMaintenanceJob::UP job;
+        if (config.getLidSpaceCompactionConfig().useBucketExecutor()) {
+            job = std::make_unique<LidSpaceCompactionJob>(
```

Comment:
_lidspace::CompactionJob_ should be instantiated here, or we must negate the expression in the if statement.

Reply:
Correct, fixed #16190
Very good observations, both you and factory came to the same conclusion :)